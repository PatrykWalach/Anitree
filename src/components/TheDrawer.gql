#import './TheDrawerViewer.gql'

query TheDrawerQuery {
  Viewer {
    id
    ...TheDrawerViewer_viewer
  }
}
