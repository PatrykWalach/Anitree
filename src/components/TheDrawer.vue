<template>
  <v-navigation-drawer :value="value" app @input="input">
    <!-- <TheDrawerSearch v-model="search" /> -->
    <TheDrawerFilters v-if="filter" />
    <v-divider></v-divider>
    <TheDrawerSettings />
  </v-navigation-drawer>
</template>

<script lang="ts">
import { Vue, Component, Prop, Emit } from 'vue-property-decorator'

import TheDrawerSettings from './TheDrawerSettings.vue'
const TheDrawerFilters = () => import('./TheDrawerFilters.vue')

import media from '../store/modules/media'

@Component({
  components: {
    TheDrawerSettings,
    TheDrawerFilters
  }
})
export default class TheDrawer extends Vue {
  @Prop()
  public readonly value!: boolean

  get filter() {
    return media.activeFilters.length
  }

  @Emit()
  public input(value: boolean) {
    return value
  }
}
</script>
