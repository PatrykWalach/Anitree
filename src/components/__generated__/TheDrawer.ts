/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: TheDrawer
// ====================================================

export interface TheDrawer_Media_title {
  readonly __typename: "MediaTitle";
  /**
   * The currently authenticated users preferred title language. Default romaji for non-authenticated
   */
  readonly userPreferred: string | null;
}

export interface TheDrawer_Media {
  readonly __typename: "Media";
  /**
   * The id of the media
   */
  readonly id: number;
  /**
   * The official titles of the media in various languages
   */
  readonly title: TheDrawer_Media_title | null;
}

export interface TheDrawer {
  /**
   * Media query
   */
  readonly Media: TheDrawer_Media | null;
}

export interface TheDrawerVariables {
  readonly id?: number | null;
}
