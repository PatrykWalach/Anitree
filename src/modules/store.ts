import * as modules from '.'
import { Store, plugin } from 'vuex-composition-api'

export const store =
  process.env.NODE_ENV === 'production'
    ? undefined
    : new Store({
        plugins: [plugin(Object.values(modules))],
      })
