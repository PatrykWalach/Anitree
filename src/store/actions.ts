export { titleActions as title } from './modules/title'

export { editActions as edit } from './modules/edit'
export { filterActions as filter } from './modules/filter'
export { shareActions as share } from './modules/share'
export { settingsActions as settings } from './modules/settings'
export { timelineActions as timeline } from './modules/timeline'
export { changesActions as changes } from './modules/changes'
