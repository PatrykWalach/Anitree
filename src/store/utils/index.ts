export * from './useCommands'
export * from './useTitle'
export * from './useEdit'
