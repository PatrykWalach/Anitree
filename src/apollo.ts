import { ApolloLink, concat } from 'apollo-link'
import { InMemoryCache, NormalizedCacheObject } from 'apollo-cache-inmemory'
import { PersistedData, PersistentStorage } from 'apollo-cache-persist/types'

import { ApolloClient } from 'apollo-client'
import { HttpLink } from 'apollo-link-http'

import { Variables } from '@/graphql/schema/media'
import Vue from 'vue'
import VueApollo from 'vue-apollo'
import { persistCache } from 'apollo-cache-persist'

import { store } from '@/store'

const link = new HttpLink({
  uri: 'https://graphql.anilist.co',
})

const middle = new ApolloLink((operation, forward) => {
  const token = store.getState().settings.token

  if (token) {
    operation.setContext({
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
  }
  if (forward) {
    return forward(operation)
  }
  return null
})

const cache = new InMemoryCache({
  cacheRedirects: {
    Query: {
      Media: (_, args, { getCacheKey }) =>
        getCacheKey({ __typename: 'Media', id: (args as Variables).id }),
    },
  },
})

// if (store.getState()settings..cacheApollo) {
persistCache({
  cache,
  storage: window.localStorage as PersistentStorage<
    PersistedData<NormalizedCacheObject>
  >,
})
// }

export const apollo = new ApolloClient({
  cache,
  link: concat(middle, link),
})

Vue.use(VueApollo)

export const apolloProvider = new VueApollo({
  defaultClient: apollo,
})
