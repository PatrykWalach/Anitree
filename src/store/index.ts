import Vue from 'vue'
import Vuex from 'vuex'

// import { getModule } from 'vuex-module-decorators'
Vue.use(Vuex)

import { ModuleMedia } from './modules/media'
import { ModuleTheme } from './modules/theme'
import { ModuleTitle } from './modules/title'
import { ModuleAuth } from './modules/auth'

export interface RootState {
  media: ModuleMedia
  theme: ModuleTheme
  title: ModuleTitle
  auth: ModuleAuth
}

export default new Vuex.Store<RootState>({})
