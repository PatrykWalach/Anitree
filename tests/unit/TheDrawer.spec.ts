import { createVariations, matchSnapshot } from './utils'
import TheDrawer from '@/components/TheDrawer.vue'
import { useMockedApollo } from './mocks/apollo'
import { useMockedStore } from './mocks/store'

describe('TheDrawer', () => {
  createVariations({
    mocks: {
      $route: () => [
        ...createVariations({
          name: () => ['search', 'home'],
        }),
        ...createVariations({
          name: () => ['timeline', 'timeline'],
          params: {
            mediaId: 1,
          },
        }),
      ],
    },
    propsData: {
      value: () => [true, false],
    },
  }).forEach((settings) =>
    matchSnapshot(TheDrawer, {
      ...settings,
      provide: { ...useMockedStore(), ...useMockedApollo() },
    }),
  )
})
