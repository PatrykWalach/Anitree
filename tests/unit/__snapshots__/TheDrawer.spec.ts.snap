// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheDrawer matches snapshot 1`] = `
<v-navigation-drawer-stub
  app="true"
  height="100vh"
  minivariant="true"
  minivariantwidth="72"
  mobilebreakpoint="1264"
  permanent="true"
  src=""
  tag="nav"
  value="true"
  width="256"
>
  <!---->
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <thedrawerviewerlogin-stub />
     
    <v-divider-stub />
  </v-list-stub>
   
  <v-list-stub
    rounded="true"
    tag="div"
    tile="true"
  >
    <v-fab-transition-stub
      mode="out-in"
      origin="center center"
    >
      <v-list-item-stub
        activeclass=""
        class="accent"
        tag="div"
      >
        <v-list-item-icon-stub>
          <v-icon-stub
            color="white"
          >
            
            tune
          
          </v-icon-stub>
        </v-list-item-icon-stub>
         
        <v-list-item-content-stub>
          <v-list-item-title-stub>
            
            Filter
          
          </v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
    </v-fab-transition-stub>
  </v-list-stub>
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          home
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Home
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          change_history
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Changes
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          settings
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Settings
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;

exports[`TheDrawer matches snapshot 2`] = `
<v-navigation-drawer-stub
  app="true"
  height="100vh"
  minivariant="true"
  minivariantwidth="72"
  mobilebreakpoint="1264"
  permanent="true"
  src=""
  tag="nav"
  value="true"
  width="256"
>
  <!---->
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <thedrawerviewerlogin-stub />
     
    <v-divider-stub />
  </v-list-stub>
   
  <v-list-stub
    rounded="true"
    tag="div"
    tile="true"
  >
    <v-fab-transition-stub
      mode="out-in"
      origin="center center"
    >
      <v-list-item-stub
        activeclass=""
        class="accent"
        exact="true"
        tag="div"
        to="[object Object]"
      >
        <v-list-item-icon-stub>
          <v-icon-stub
            color="white"
          >
            
            search
          
          </v-icon-stub>
        </v-list-item-icon-stub>
         
        <v-list-item-content-stub>
          <v-list-item-title-stub>
            
            Search
          
          </v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
    </v-fab-transition-stub>
  </v-list-stub>
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          home
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Home
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          change_history
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Changes
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          settings
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Settings
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;

exports[`TheDrawer matches snapshot 3`] = `
<v-navigation-drawer-stub
  app="true"
  height="100vh"
  minivariant="true"
  minivariantwidth="72"
  mobilebreakpoint="1264"
  permanent="true"
  src=""
  tag="nav"
  value="true"
  width="256"
>
  <!---->
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <thedrawerviewerlogin-stub />
     
    <v-divider-stub />
  </v-list-stub>
   
  <v-list-stub
    rounded="true"
    tag="div"
    tile="true"
  >
    <v-fab-transition-stub
      mode="out-in"
      origin="center center"
    >
      <v-list-item-stub
        activeclass=""
        class="accent"
        tag="div"
      >
        <v-list-item-icon-stub>
          <v-icon-stub
            color="white"
          >
            
            sort
          
          </v-icon-stub>
        </v-list-item-icon-stub>
         
        <v-list-item-content-stub>
          <v-list-item-title-stub>
            
            Sort
          
          </v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
    </v-fab-transition-stub>
  </v-list-stub>
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          home
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Home
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          change_history
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Changes
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          settings
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Settings
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;

exports[`TheDrawer matches snapshot 4`] = `
<v-navigation-drawer-stub
  app="true"
  height="100vh"
  minivariant="true"
  minivariantwidth="72"
  mobilebreakpoint="1264"
  permanent="true"
  src=""
  tag="nav"
  value="true"
  width="256"
>
  <!---->
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <thedrawerviewerlogin-stub />
     
    <v-divider-stub />
  </v-list-stub>
   
  <v-list-stub
    rounded="true"
    tag="div"
    tile="true"
  >
    <v-fab-transition-stub
      mode="out-in"
      origin="center center"
    >
      <v-list-item-stub
        activeclass=""
        class="accent"
        tag="div"
      >
        <v-list-item-icon-stub>
          <v-icon-stub
            color="white"
          >
            
            edit
          
          </v-icon-stub>
        </v-list-item-icon-stub>
         
        <v-list-item-content-stub>
          <v-list-item-title-stub>
            
            Edit
          
          </v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
    </v-fab-transition-stub>
  </v-list-stub>
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          home
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Home
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          change_history
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Changes
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          settings
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Settings
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;

exports[`TheDrawer matches snapshot 5`] = `
<v-navigation-drawer-stub
  app="true"
  height="100vh"
  minivariant="true"
  minivariantwidth="72"
  mobilebreakpoint="1264"
  permanent="true"
  src=""
  tag="nav"
  width="256"
>
  <!---->
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <thedrawerviewerlogin-stub />
     
    <v-divider-stub />
  </v-list-stub>
   
  <v-list-stub
    rounded="true"
    tag="div"
    tile="true"
  >
    <v-fab-transition-stub
      mode="out-in"
      origin="center center"
    >
      <v-list-item-stub
        activeclass=""
        class="accent"
        tag="div"
      >
        <v-list-item-icon-stub>
          <v-icon-stub
            color="white"
          >
            
            tune
          
          </v-icon-stub>
        </v-list-item-icon-stub>
         
        <v-list-item-content-stub>
          <v-list-item-title-stub>
            
            Filter
          
          </v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
    </v-fab-transition-stub>
  </v-list-stub>
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          home
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Home
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          change_history
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Changes
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          settings
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Settings
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;

exports[`TheDrawer matches snapshot 6`] = `
<v-navigation-drawer-stub
  app="true"
  height="100vh"
  minivariant="true"
  minivariantwidth="72"
  mobilebreakpoint="1264"
  permanent="true"
  src=""
  tag="nav"
  width="256"
>
  <!---->
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <thedrawerviewerlogin-stub />
     
    <v-divider-stub />
  </v-list-stub>
   
  <v-list-stub
    rounded="true"
    tag="div"
    tile="true"
  >
    <v-fab-transition-stub
      mode="out-in"
      origin="center center"
    >
      <v-list-item-stub
        activeclass=""
        class="accent"
        exact="true"
        tag="div"
        to="[object Object]"
      >
        <v-list-item-icon-stub>
          <v-icon-stub
            color="white"
          >
            
            search
          
          </v-icon-stub>
        </v-list-item-icon-stub>
         
        <v-list-item-content-stub>
          <v-list-item-title-stub>
            
            Search
          
          </v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
    </v-fab-transition-stub>
  </v-list-stub>
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          home
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Home
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          change_history
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Changes
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          settings
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Settings
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;

exports[`TheDrawer matches snapshot 7`] = `
<v-navigation-drawer-stub
  app="true"
  height="100vh"
  minivariant="true"
  minivariantwidth="72"
  mobilebreakpoint="1264"
  permanent="true"
  src=""
  tag="nav"
  width="256"
>
  <!---->
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <thedrawerviewerlogin-stub />
     
    <v-divider-stub />
  </v-list-stub>
   
  <v-list-stub
    rounded="true"
    tag="div"
    tile="true"
  >
    <v-fab-transition-stub
      mode="out-in"
      origin="center center"
    >
      <v-list-item-stub
        activeclass=""
        class="accent"
        tag="div"
      >
        <v-list-item-icon-stub>
          <v-icon-stub
            color="white"
          >
            
            sort
          
          </v-icon-stub>
        </v-list-item-icon-stub>
         
        <v-list-item-content-stub>
          <v-list-item-title-stub>
            
            Sort
          
          </v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
    </v-fab-transition-stub>
  </v-list-stub>
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          home
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Home
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          change_history
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Changes
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          settings
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Settings
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;

exports[`TheDrawer matches snapshot 8`] = `
<v-navigation-drawer-stub
  app="true"
  height="100vh"
  minivariant="true"
  minivariantwidth="72"
  mobilebreakpoint="1264"
  permanent="true"
  src=""
  tag="nav"
  width="256"
>
  <!---->
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <thedrawerviewerlogin-stub />
     
    <v-divider-stub />
  </v-list-stub>
   
  <v-list-stub
    rounded="true"
    tag="div"
    tile="true"
  >
    <v-fab-transition-stub
      mode="out-in"
      origin="center center"
    >
      <v-list-item-stub
        activeclass=""
        class="accent"
        tag="div"
      >
        <v-list-item-icon-stub>
          <v-icon-stub
            color="white"
          >
            
            edit
          
          </v-icon-stub>
        </v-list-item-icon-stub>
         
        <v-list-item-content-stub>
          <v-list-item-title-stub>
            
            Edit
          
          </v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
    </v-fab-transition-stub>
  </v-list-stub>
   
  <v-list-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          home
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Home
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          change_history
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Changes
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      color="primary"
      exact="true"
      tag="div"
      to="[object Object]"
    >
      <v-list-item-icon-stub>
        <v-icon-stub>
          settings
        </v-icon-stub>
      </v-list-item-icon-stub>
       
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          
          Settings
        
        </v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;
